dcos_test_utils/gce.py

```diff
@@ -0,0 +1,247 @@
+        if errors:
+            raise Exception('The deployment you are accessing contains errors:' + str(errors))
+        return response
+
+    def delete(self) -> dict:
+        response = self.gce_wrapper.delete_deployment(self)
+        log.debug(response)
+        return response
+
+    def _check_status(response):
+        '''
+        Checks the status of the deployment until it is done or has failed
+        :param response : <dict> http response containing info about the deployment
+        :return: <boolean> whether to continue checking the status of the deployment (True) or not (False)
+        '''
+        status = response['operation']['status']
+        if status == 'DONE':
+            return False
+        elif status == 'RUNNING' or status == 'PENDING':
+            log.debug('Waiting for deployment')
+            return True
+        else:
+            raise Exception('Deployment failed with response: ' + str(response))
+
+    @retry(wait_fixed=2000, retry_on_result=_check_status)
+    def wait(self) -> dict:
+        response = self.get_info()
+        log.debug(response)
+        return response
+
+    def allow_all_ports(self) -> dict:
+        body = {
+            "kind": "compute#firewall",
+            "name": "allow all ports",
+            "description": "allow all ports",
+            "network": "global/networks/default",
+            "sourceRanges": [
+                "0.0.0.0/0"
+            ],
+            "allowed": [
+                {
+                    "IPProtocol": "tcp",
+                    "ports": [
+                        "0-65535"
+                    ]
+                },
+                {
+                    "IPProtocol": "udp",
+                    "ports": [
+                        "0-65535"
+                    ]
+                },
+                {
+                    "IPProtocol": "icmp"
+                }
+            ]
+        }
+
+        response = self.gce_wrapper.add_firewall_rule(body)
+        log.debug(response)
+        return response
+
+
+def catch_http_exceptions(f):
+    '''
+    Runs the function "f" and raises an exception with a custom help message if a google HttpError with status 404 or
+        409 is caught
+    :param f: function to be executed
+    :return: return value of function "f"
+    '''
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if e.resp.status == 404:
+                log.exception("The resource you are trying to access doesn't exist")
+            elif e.resp.status == 409:
+                log.exception('''The specified resources exist and might be under an active operation
+                                   (operation conflict)''')
+            raise e
+
+    return handle_exception
+
+
+class GceWrapper:
+    @catch_http_exceptions
+    def __init__(self, credentials_dict):
+        credentials = GoogleCredentials.get_application_default()
```

Comment:
I don't think this helps me - it encapsulates the credentials object creation inside the google.cloud.client.Client class and returns me an instance of that class (which I don't need). But I think this might to do it: 
`If using a JSON file:
from oauth2client.service_account import ServiceAccountCredentials

scopes = ['https://www.googleapis.com/auth/sqlservice.admin']

credentials = ServiceAccountCredentials.from_json_keyfile_name(
    '/path/to/keyfile.json', scopes)`

from : https://developers.google.com/api-client-library/python/auth/service-accounts